rtmp-stream.go

```diff
@@ -98,6 +114,67 @@ func (f *RtmpStreamFactory) startStream(conn rtmp.ClientConn, streamName string)
 	}
 }
 
+func (f *RtmpStreamFactory) ParseURLArgument(urlArg string) (string, []*url.URL, error) {
+	var host string
+	var urls []*url.URL
+	var r = regexp.MustCompile(`{{(?P<min>[1-9][0-9]*) (?P<max>[1-9][0-9]*)}}`) // {{123 456}}
+
+	var regexString = "'{{(?P<min>[1-9][0-9]*) (?P<max>[1-9][0-9]*)}}'"
+	var regexInfo = fmt.Sprintf("Use regex that matches patter %v", regexString)
+
+	if r.MatchString(urlArg) { // URL is a template.
+		log.Infoln("Processing template URL %v with regex matching. (Used regex: '%v')", urlArg, regexString)
+		match := r.FindStringSubmatch(urlArg)
```

Comment:
The regex is matched a second time here. Instead of using MatchString() and FindStringSubmatch separately, ues the result of FindStringSubmatch to find out IF there was a match.

Reply:
All other comments are fixed. From performance perspective I  agree with this one. However, it is not function that  is called often and the way it is now is better readable from my point of view.